### PR TITLE
Add connector to support Niconico

### PR DESCRIPTION
**Pull request by takagiy**
**Describe the changes you made**
This request is going to add support for Niconico (www.nicovideo.jp), a popular video streaming service especially in japan, by setting up a new connector.

**Additional context**
RE: #1932

### CODE REVIEW

**Comment by takagiy** [comment]
Oh, I missed `src/core/content/util.js`. Maybe, should I do some refactoring?

**Review by alexesprit — CHANGES_REQUESTED** [review]

**Comment by alexesprit** [comment]
> Oh, I missed src/core/content/util.js (looks useful for filtering). Maybe, should I do some refactoring?

What do you want to use from this module?

File: src/connectors/nicovideo.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	const extraInfoPattern = /\s*(\/[^/]+|／[^／]+|[^\da-zA-Z]-[^\da-zA-Z][^-]+|\sfeat\..+|\sft\..+)$/i;
+	// Or, something like "Original Song『Track Name』MV"
+	const titlePattern = /「(.+)」|『(.+)』|'(.+)'|"(.+)"/;
+
+	const match = text.match(titlePattern);
+	if (match) {
+		return match[1] || match[2] || match[3] || match[4];
+	}
+
+	return text.replaceAll(decorationPattern, '').replace(extraInfoPattern, '');
```
[the remark this answers is not in the document]
Comment (takagiy) [reply src/connectors/nicovideo.js]: Thanks you.
fix: b7da50a

**Comment by takagiy** [comment]
@alexesprit I'll try to replace [this matching](https://github.com/takagiy/web-scrobbler/blob/b7da50aaea379aad6807e69ed0cd59f0a003601b/src/connectors/nicovideo.js#L32-L33) with `Util.splitString()` or `Util.splitArtistTrack` if you prefer it. But I think the regex pattern can do bit more complex (e.g. requiring some kind of whitespace before `feat.`, treating `-` surrounded by non-alphanumerics as separators, but not those that aren't and using `/` at the most right as separators).

**Comment by alexesprit** [comment]
It's fine to use regular expressions here.

**Comment by takagiy** [comment]
I see, I'll leave it as it is. Thank you.

**Comment by takagiy** [comment]
I noticed that when I switched my language setting to Japanese, Niconico shows artist name with -さん suffix like "Artist さん". I'll modify the filter to remove that.

**Review by alexesprit — APPROVED** [review]

**Pull request merged**